packages/m2m-gateway/src/services/purposeService.ts

```diff
@@ -1,8 +1,43 @@
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
+import { WithLogger } from "pagopa-interop-commons";
+import { toM2MGatewayApiPurpose } from "../api/purposeApiConverter.js";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
 
 export type PurposeService = ReturnType<typeof purposeServiceBuilder>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function purposeServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function purposeServiceBuilder(clients: PagoPAInteropBeClients) {
+  return {
+    getPurposes: async (
+      { logger, headers }: WithLogger<M2MGatewayAppContext>,
+      queryParams: m2mGatewayApi.GetPurposesQueryParams
```

Comment:
Minor, for consistency, context is usually the last param we pass, invert them
```suggestion
         queryParams: m2mGatewayApi.GetPurposesQueryParams,
         { logger, headers }: WithLogger<M2MGatewayAppContext>
```